feature_engine/discretisation/base_discretiser.py

```diff
@@ -0,0 +1,120 @@
+# Authors: Morgan Sell <morganpsell@gmail.com>
+# License: BSD 3 clause
+
+from typing import List, Optional, Union
+
+import pandas as pd
+
+from sklearn.utils.validation import check_is_fitted
+
+from feature_engine.dataframe_checks import (
+    _check_contains_na,
+    _check_input_matches_training_df,
+    _is_dataframe,
+)
+
+from feature_engine.base_transformers import BaseNumericalTransformer
+from feature_engine.dataframe_checks import _check_contains_na
+
+import warnings
```

Comment:
this one should be at the top. 

If you install isort (`pip install isort`), then you can run `isort myfile` and that will sort the imports according to pep8

Here be extra careful, if you run `isort feature_engine`, it might reorganise the imports of many files, and we do not want files not relevant to this PR added here. 

Final note, we normally run isort after we added all relevant code. Like with black and flake8